packages/react-components/src/components/Search/Search.spec.tsx

```diff
@@ -95,43 +188,56 @@ describe('<Search> component', () => {
     expect(queryByPlaceholderText(/Custom placeholder/i)).toBeVisible();
   });
 
-  it('should render clear icon if value is given', () => {
-    const { getByRole, queryByTestId } = renderComponent({
+  it('should render clear icon if value is given by the user', () => {
+    const { getByRole, queryByTestId, rerender } = renderComponent({
       ...defaultProps,
     });
+    const text = 'test value';
 
     expect(queryByTestId(`${baseClass}-clear-icon`)).toBeFalsy();
-    fireEvent.change(getByRole('textbox'), { target: { value: 'test' } });
+    userEvent.type(getByRole('textbox'), text);
+
+    rerender(
+      rerenderedComponent({
+        ...defaultProps,
+        value: text,
+      })
+    );
+
     expect(queryByTestId(`${baseClass}-clear-icon`)).toBeVisible();
   });
 
   it('should call onChange if input value change', () => {
     const onChangeFunction = vi.fn();
-    const { getByRole } = renderComponent({
-      ...defaultProps,
-      onChange: onChangeFunction,
-    });
-
-    fireEvent.change(getByRole('textbox'), { target: { value: 'test' } });
-    expect(onChangeFunction).toBeCalledWith('test');
+    const SearchWrapper = () => {
+      const [value, setValue] = React.useState('');
+      return (
+        <SearchInput
+          value={value}
+          onChange={(v) => {
+            setValue(v);
+            onChangeFunction(v);
+          }}
+        />
+      );
+    };
+
+    const { getByRole } = render(<SearchWrapper />);
+    const text = 'test value';
+
+    userEvent.type(getByRole('textbox'), text, { delay: 0 });
+    expect(onChangeFunction).toHaveBeenCalledWith(text);
```

Comment:
Since this test is for a controlled component, we need to render it along with the external state, if we want to use `userEvent` to simulate a user entering text.